x-pack/plugins/apm/public/components/app/TransactionDetails/Distribution/index.tsx

```diff
@@ -47,191 +53,226 @@ export function getFormattedBuckets(buckets: IBucket[], bucketSize: number) {
   );
 }
 
+const getFormatYShort = (transactionType: string | undefined) => (
+  t: number
+) => {
+  return i18n.translate(
+    'xpack.apm.transactionDetails.transactionsDurationDistributionChart.unitShortLabel',
+    {
+      defaultMessage:
+        '{transCount} {transType, select, request {req.} other {trans.}}',
+      values: {
+        transCount: t,
+        transType: transactionType
+      }
+    }
+  );
+};
+
+const getFormatYLong = (transactionType: string | undefined) => (t: number) => {
+  return transactionType === 'request'
+    ? i18n.translate(
+        'xpack.apm.transactionDetails.transactionsDurationDistributionChart.requestTypeUnitLongLabel',
+        {
+          defaultMessage:
+            '{transCount, plural, =0 {# request} one {# request} other {# requests}}',
+          values: {
+            transCount: t
+          }
+        }
+      )
+    : i18n.translate(
+        'xpack.apm.transactionDetails.transactionsDurationDistributionChart.transactionTypeUnitLongLabel',
+        {
+          defaultMessage:
+            '{transCount, plural, =0 {# transaction} one {# transaction} other {# transactions}}',
+          values: {
+            transCount: t
+          }
+        }
+      );
+};
+
 interface Props {
   location: Location;
   distribution?: ITransactionDistributionAPIResponse;
   urlParams: IUrlParams;
 }
 
-export class TransactionDistribution extends Component<Props> {
-  public formatYShort = (t: number) => {
-    return i18n.translate(
-      'xpack.apm.transactionDetails.transactionsDurationDistributionChart.unitShortLabel',
-      {
-        defaultMessage:
-          '{transCount} {transType, select, request {req.} other {trans.}}',
-        values: {
-          transCount: t,
-          transType: this.props.urlParams.transactionType
-        }
-      }
-    );
-  };
+export const TransactionDistribution: FunctionComponent<Props> = (
+  props: Props
+) => {
+  const {
+    location: sourceLocation,
```

Comment:
Why rename to `sourceLocation`? Is that to avoid overwriting `window.location`? 

As an aside, I've noticed an issue where we (by mistake) access `window.location` when writing `location` - I think we can solve this with the eslitn rule `no-restricted-globals`

Reply:
Oh, I see. You declare `location` below from the ref.

Reply:
@sqren I think that rule is already in use:

![image](https://user-images.githubusercontent.com/352732/57403378-de202000-71d9-11e9-85e3-9bc129b8c636.png)

Reply:
Interesting, I only get those warnings in js-files but not in ts.